_posts/2021-05-14-bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton.md

```diff
@@ -0,0 +1,612 @@
+---
+title: "Bring GitOps to your OpenFaaS functions with Argo CD - The missing piece of the pipeline: Tekton"
+description: "In this post you'll learn how to combine these two technologies ArgoCD and Tekton to build fully automated CI/CD pipeline which can be triggerred by Github to manage OpenFaaS functions based on GitOps principles in your local environment using a Cloud Native Tunnel called Inlets"
+date: 2021-05-14
+image: /images/bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton/tekton_argocd_power.png
+categories:
+- arkade
+- kubectl
+- argocd
+- tekton
+- gitops
+- openfaas-operator
+- helm
+author_staff_member: developer-guy
+dark_background: true
+
+---
+
+In this post you'll learn how to combine these two technologies ArgoCD and Tekton to build fully automated CI/CD pipeline which can be triggerred by Github to manage OpenFaaS functions based on GitOps principles in your local environment using a Cloud Native Tunnel called Inlets
+
+## Introduction
+
+We talked about _How we can bring GitOps principles to management of OpenFaaS functions_ in the previous blog post. If you haven't read it yet, you can follow this [link](https://www.openfaas.com/blog/bring-gitops-to-your-openfaas-functions-with-argocd/), because it might help you to understand the all of the pieces of CI/CD pipeline we want to build here. In this guide, we'll add CI (Continious Integration) part to our pipeline by using _Tekton_. After that, to be able to trigger this pipeline based on _Github_ events, we'll use an another component of [Tekton](https://tekton.dev) called [Tekton Trigger](https://tekton.dev/docs/triggers/). We will do this demo using KinD on a kubernetes running locally. Because of we are in a private network, we have to listen events that send by Github to trigger our Tekton Pipeline. So we need to find a way to susbcribe those events, and this is where Tekton Triggers comes into the picture. So, we said that everyhing is in local, so, we should open our local services to the internet, Github in this case, to be able Github send events to our event listener, and this is where Inlets, a Cloud Native Tunnel, comes in to the picture. At the end of this tutorial, we'll have a pipeline like the following:
```

Comment:
Try to break this up into two smaller paragraphs so that it's more likely people will read it.

Add a link for inlets. Add a link for GitHub and write it as "GitHub" (not Github)